src/lib/Masonry.svelte

```diff
@@ -45,7 +48,7 @@
   {#each itemsToCols as col}
     <div class="col {columnClass}" style="gap: {gap}px; max-width: {maxColWidth}px;">
       {#if animate}
-        {#each col as [item, idx] (getId(item))}
+        {#each col as [item, idx] (isIterator ? id(item) : item[id])}
```

Comment:
I considered this solution in #11 but like you said it doesn't type check. If we can figure out a way to not throw compiler errors, I'm happy to change the API.